server/mdm/apple/apple_mdm.go

```diff
@@ -347,7 +347,13 @@ func (d *DEPService) RunAssigner(ctx context.Context) error {
 			return err
 		}
 	}
-	return d.syncer.Run(ctx)
+	if err := d.syncer.Run(ctx); err != nil {
+		return err
+	}
+
+	// TODO: maybe process cooldowns here instead of the integrations cron?
```

Comment:
keep in mind that this runs every 30 seconds, which might be too much considering the cooldown is 1 hour. Having that said it's a completely valid option.

Reply:
Yep, I think the integrations cron strikes the right balance but wanted to bring up the possibility. I'll remove the TODO here.